### PR TITLE
Add parent .model to ModelTypeProperty

### PR DESCRIPTION
**Pull request by nguerrera**
This add a .model to model properties just as there is a .interface on operations.

The property is optional because it is currently possible to get a model property from getTypeForNode without checking the parent model and we don't memoize getTypeForNode(propertyNode). I started down a road of addressing this, but it got complicated and I thought I'd put this out with parity between operation interface and model properties first, and we can see if that is even a problem worth solving. Even if so, I'd prefer to move on to other parts of metadata work in the short term before going any deeper on that.

### CODE REVIEW

File: packages/compiler/core/checker.ts [hunk 1]
```diff
@@ -796,8 +792,16 @@ export function createChecker(program: Program): Checker {
...
     const optionTypes = node.options.map(getTypeForNode);
-
     const properties = new Map<string, ModelTypeProperty>();
+
+    const intersection: ModelType = createType({
+      kind: "Model",
+      node,
+      name: "",
+      properties: properties,
+      decorators: [], // could probably include both sets of decorators here...
```
Comment (timotheeguerin) [inline packages/compiler/core/checker.ts:802]: See this was already there but is there something we need to update here? or should this comment be removed

**Review by timotheeguerin — APPROVED** [review]

File: packages/compiler/core/checker.ts (hunk 1 above)
Comment (nguerrera) [reply packages/compiler/core/checker.ts:802]: I was just staring at this too. @bterlson thoughts?

Comment (bterlson) [reply packages/compiler/core/checker.ts:802]: Sorry for crappy comments. I think this comment is very outdated and could be deleted. At some point before we decided & was sugar for ..., we might have copied decorators from the sources at this point. Comment can be removed.

File: packages/openapi3/src/openapi.ts [hunk 2]
```diff
@@ -626,7 +626,7 @@ function createOAPIEmitter(program: Program, options: OpenAPIEmitterOptions) {
   }
 
   function getParameterKey(property: ModelTypeProperty, param: any) {
-    const parent = program.checker!.getTypeForNode(property.node.parent!) as ModelType;
+    const parent = property.model!;
```
Comment (bterlson) [inline packages/openapi3/src/openapi.ts:629]: noice.

**Review by bterlson — APPROVED** [review]

**Pull request merged**